### ISSUE TITLE
Xtext LSP does not work with files with Cyrillic letters in path.

### ISSUE THREAD

**Issue by xcariba** [eclipse/xtext-core#396]
I'm currently working with Eclipse Che and Xtext LSP. I've built 2.12 LSP bundle and it works fine with Che, but it does not support Cyrillic letters. 
LSP Server recieves UTF String with file uri (file://<path>/<cyrillic name>.mydsl) and converts it to URI in `UriExtensions` extension class. URI encodes path, but server does not decode it properly and LSP does not work for such files.
Encoding problem accures on `Paths.get(uri)`
Reproduces on Linux (Fedora 26, JavaSE 1.8 111).

**Comment by xcariba** [comment]
**Proposed solution:**
in `org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/UriExtensions.xtend`
`def String toPath(java.net.URI uri) {
		try {
			val path = Paths.get(uri.toPath)
			return URLDecoder.decode(path.toUri.toString, "UTF-8");
		} catch (FileSystemNotFoundException e) {
			return uri.toString
		}
	}`

**Comment by xcariba** [comment]
**Proposed solution:**
in `org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/UriExtensions.xtend`
```xtend
def String toPath(java.net.URI uri) {
    try {
        val path = Paths.get(uri.path)
        return URLDecoder.decode(path.toUri.toString, "UTF-8");
    } catch (FileSystemNotFoundException e) {
        return uri.toString
    }
}
```

**Comment by svenefftinge** [comment]
Could you provide a PR that includes a test?

**Comment by xcariba** [comment]
@svenefftinge I will provide test. 
Also there is a problem with spaces in paths, since `URI.create(..)` throws `IllegalArgumentException` when there are spaces in argument. Found this bug in Eclipse Che with xtext-based LSP. Server gets UTF-8 uri path in `file:///projects/Project/File name.ext` format, which is already decoded, then in encodes it in `java.net.URI` and passes it to nio Path, converts it to string and then passes it to EMF`s `URI`. I`m not actually sure why can`t you use just EMF `URI.createUri(..)`.

**Comment by xcariba** [comment]
@svenefftinge PR: #403. I've removed `java.net.URI` to fix this issue and used EMF URI instead.

**Comment by xcariba** [comment]
Fixed in #403
